library/core/src/array/mod.rs

```diff
@@ -29,23 +29,21 @@ pub use iter::IntoIter;
 /// # Example
 ///
 /// ```rust
-/// #![feature(array_from_fn)]
-///
-/// let array = core::array::from_fn(|i| i);
+/// let mut idx: usize = 0;
+/// let array = core::array::from_fn(|| {
+///   let rslt = idx;
+///   idx = idx.wrapping_add(1);
```

Comment:
I find this wrapping_add distracting. Array sizes in Rust are `usize`, so any array has at most `usize::MAX` elements, so `from_fn` will run the closure at most `usize::MAX` times, which means incrementing `idx` from 0 at most `usize::MAX` times by 1, which does not overflow. Recommend `+=` instead.

Reply:
Done